Feature/timetracker

jonasfranz opened the pull request:
Adding feature tracking API implementation.

See go-gitea/gitea/pull/2211

jonasfranz committed:
Adding time tracker api

jonasfranz committed:
Adding swagger support
Adding GetMyTrackedTimes

Signed-off-by: Jonas Franz <email@jfdev.de>

jonasfranz committed:
Adding swagger support
Adding GetMyTrackedTimes
Fixed package collision
+ gofmt

Signed-off-by: Jonas Franz <email@jfdev.de>

jonasfranz committed:
Adding documentation

Signed-off-by: Jonas Franz <email@jfdev.de>

JonasFranzDEV commented:
Example: https://gist.github.com/JonasFranzDEV/b9ba83380e785d9f5fd7cd035a4dace8

lafriks commented on gitea/issue_tracked_time.go:
```diff
@@ -0,0 +1,62 @@
+
+// TrackedTimes represent a list of tracked times
+// swagger:response TrackedTimes
+type TrackedTimes []*TrackedTime
```
Why this type is needed if it is not used anywhere?

jonasfranz replied:
It is used for Swagger

lafriks replied:
Can it then be used also as return type for these functions?

jonasfranz replied:
@lafriks I replaced []*TrackedTime with TrackedTimes

jonasfranz committed:
Replacing []*TrackedTime with TrackedTimes

Signed-off-by: Jonas Franz <email@jfdev.de>

jonasfranz committed:
Adding GetRepoTrackedTimes
Changing endpoint of GetUserTrackedTimes
See go-gitea/gitea/pull/2211
(Applying changes from https://github.com/go-gitea/gitea/pull/2211/commits/078f59fbd8f84871ab77553afee78a928ac11528)

Signed-off-by: Jonas Franz <email@jfdev.de>

lafriks commented:
LGTM

appleboy commented:
LGTM
